### DISCUSSION TITLE
It works - but why?

### DISCUSSION THREAD

**Discussion by SprickW** [jhuckaby/Cronicle#627] [Q&A]
Just installed a primary and a worker.

Ran "sudo apt update && sudo apt dist-upgrade -y" on all servers. And it worked.

But why? 

You need to authenticate and to be authorized on every server. But I did not find any documentation on authentication and authorization in the docs.

Any hints?

Some commands have to be run as a specific user on a specific server, as access right to a database server are associated with this account. How can this be accomplished?

Greetings Wolfgang

**Comment by jhuckaby** [comment]
Cronicle is not part of the "apt" system.  It's an independent software package.  For upgrading Cronicle, see the docs here: https://github.com/jhuckaby/Cronicle/blob/master/docs/CommandLine.md#upgrading-cronicle

You can enter a custom UID/GID for Plugins to run as.  Clone the shell plugin for example.

**Reply by SprickW** [reply]
Sorry for the misunderstanding. Using "sudo" and "apt" was just meant as an example.

I want to understand how authentication and authorization works when cronicle workers execute shell commands on some server. Is everything run as root?

I found some hint that cloning the shell plugin and specifying a user (and a working directory) for the new plugin might do the job. That hint also showed that a secret (password?) could be specified. But my implementation doesn't allow specifying a secret. How does the worker execute commands using plugins that specify a user account?

Please understand, that security is a mayor issue when using cronicle even in a small business relying on its IT infrastructure.

Kind regards Wolfgang

**Comment by dheise-rheinenergie-trading-com** [comment]
the master and every worker is normally running with root permissions, which makes it possible to run scripts under the root account or any other account which is available on the machines.
